### PR TITLE
BFD-1762 - Flyway script to migrate Hospice claims data

### PR DESCRIPTION
**Pull request by mackec1**
**JIRA Ticket:**
[BFD-1762](https://jira.cms.gov/browse/BFD-17562)

**User Story or Bug Summary:**
Flyway migration script to migrate Hospice claims data.
- migrate hospice_claims to hospice_claims_new
- migrate hospice_claim_lines to hospice_claim_lines_new
- add appropriate data column constraints based on current schema
- add appropriate table indexes based on current schema

### What Does This PR Do?

- migrate hospice_claims to hospice_claims_new
- migrate hospice_claim_lines to hospice_claim_lines_new
- add appropriate data column constraints based on current schema
- add appropriate table indexes based on current schema
- The following db columns were redefined from NUMERIC to more appropriate data type(s):
   - from: clm_line_num numeric                         : changed to smallint

### What Should Reviewers Watch For?
- appropriate data types
- flyways script is well documented

If you're reviewing this PR, please check for these things in particular:
- flyway scripts works for both HSQL and PSQL
- scripts are documented and well structured.

### What Security Implications Does This PR Have?

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* No prerequisites; **however**, it is expected that other flyway scripts may be merged into master prior to this script; this means that the flyway script name (versioning) will need need to be modified to ensure Flyway processing.

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V69__new_schema_hospice_claims_migration.sql [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+--
+--       Current db data values for changed columns
+--      +-----------------------+--------+------+-------+
+--      |  db Column            |   Max  |  Min | Scale |
+--      +-----------------------+--------+------+-------+
+--      | CLM_UTLZTN_DAY_CNT    |    240 | -87  |   0   |
+--      | BENE_HOSPC_PRD_CNT    |      3 |   0  |   0   |
+--      | CLM_LINE_NUM          |    450 |   1  |   0   |
+--      | REV_CNTR_UNIT_CNT     | 904404 |   0  |   0   |
+--      +-----------------------+--------+------+-------+
```
Comment (christrotteradhoc) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V69__new_schema_hospice_claims_migration.sql:25]: apart from clm_line_num, I think the additional scope is a risk that we should not take right now

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V69__new_schema_hospice_claims_migration.sql [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+${logic.psql-only} alter table public.hospice_claim_lines_new
+${logic.psql-only}    alter column clm_id                       SET NOT NULL,
+${logic.psql-only}    alter column clm_line_num                 SET NOT NULL,
+${logic.psql-only}    alter column rev_cntr                     SET NOT NULL,
+${logic.psql-only}    alter column rev_cntr_unit_cnt            SET NOT NULL,
+${logic.psql-only}    alter column rev_cntr_tot_chrg_amt        SET NOT NULL,
+${logic.psql-only}    alter column rev_cntr_rate_amt            SET NOT NULL,
+${logic.psql-only}    alter column rev_cntr_bene_pmt_amt        SET NOT NULL,
+${logic.psql-only}    alter column rev_cntr_pmt_amt_amt         SET NOT NULL,
+${logic.psql-only}    alter column  rev_cntr_prvdr_pmt_amt      SET NOT NULL;
```
Comment (christrotteradhoc) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V69__new_schema_hospice_claims_migration.sql:502]: extreme nitpick on the extra space (sorry)

**Review by christrotteradhoc — COMMENTED** [review]
It checks out with the spreadsheet and I'm ready to approve once we know which one of these is going in first

**Review by christrotteradhoc — APPROVED** [review]

**Review by lsmitchell — APPROVED** [review]
Looks good, I spot checked a couple fields I was wondering about the conversion of and they look to be integers in the real data as far as I can see. (I'm sure you're doing your own checks as well anyway)

**Pull request merged**